src/connector/src/source/cdc/source/reader.rs

```diff
@@ -119,38 +119,56 @@ impl CommonSplitReader for CdcSplitReader {
             properties.insert("table.name".into(), table_name);
         }
 
-        let cdc_stream = cdc_client
-            .start_source_stream(
-                self.source_id,
-                self.conn_props.get_source_type_pb()?,
-                self.start_offset,
-                properties,
-                self.snapshot_done,
-            )
-            .await
-            .inspect_err(|err| tracing::error!("connector node start stream error: {}", err))?;
-        pin_mut!(cdc_stream);
-        #[for_await]
-        for event_res in cdc_stream {
-            match event_res {
-                Ok(GetEventStreamResponse { events, .. }) => {
-                    if events.is_empty() {
-                        continue;
-                    }
-                    let mut msgs = Vec::with_capacity(events.len());
-                    for event in events {
-                        msgs.push(SourceMessage::from(event));
-                    }
-                    yield msgs;
-                }
-                Err(e) => {
-                    return Err(anyhow!(
-                        "Cdc service error: code {}, msg {}",
-                        e.code(),
-                        e.message()
-                    ))
-                }
+        let (tx, mut rx) = mpsc::channel(1024);
+        let tx: Box<GetEventStreamJniSender> = Box::new(tx);
+
+        JVM.as_ref()?;
+
+        let get_event_stream_request = GetEventStreamRequest {
+            source_id: self.source_id,
+            source_type: self.conn_props.get_source_type_pb()? as _,
+            start_offset: self.start_offset.unwrap_or_default(),
+            properties,
+            snapshot_done: self.snapshot_done,
+        };
+
+        std::thread::spawn(move || {
+            let mut env = JVM
+                .as_ref()
+                .unwrap()
+                .attach_current_thread_as_daemon()
+                .unwrap();
+
+            let get_event_stream_request_bytes = env
+                .byte_array_from_slice(&Message::encode_to_vec(&get_event_stream_request))
+                .unwrap();
+
+            let channel_ptr = Box::into_raw(tx) as i64;
+
+            let _ = env
+                .call_static_method(
+                    "com/risingwave/connector/source/core/JniDbzSourceHandler",
+                    "runJniDbzSourceThread",
+                    "([BJ)V",
+                    &[
+                        JValue::Object(&get_event_stream_request_bytes),
+                        JValue::from(channel_ptr),
+                    ],
+                )
+                .inspect_err(|e| tracing::error!("jni call error: {:?}", e))
+                .unwrap();
```

Comment:
Here we'd better log out the error instead of unwrap?

Reply:
We have a try-catch on the Java side for those parts that might throw an expected error like debezium engine, so it should always be safe to unwrap. For those unexpected parts e.g. class not found, I think we'd better panic ASAP.

Reply:
If we panic, the log may not be printed due to buffering. 🥵